File: consecutive_1s_not_allowed.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;
#define ll long long

// } Driver Code Ends
// User function template for C++
class Solution {
 public:
  // #define ll long long
  ll countStrings(int n) {
    // code here
    const int mod = 1e9 + 7;
    vector<ll> dp1(n), dp2(n);
    dp1[0] = 1;
    dp2[0] = 1;
    for (int i = 1; i < n; ++i) {
      dp1[i] = (dp1[i - 1] + dp2[i - 1]) % mod;
      dp2[i] = dp1[i - 1];
    }
    return (dp1[n - 1] + dp2[n - 1]) % mod;
  }
};

//{ Driver Code Starts.

int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    Solution ob;
    auto ans = ob.countStrings(n);
    cout << ans << "\n";
  }
  return 0;
}
// } Driver Code Ends

File: dp/consecutive_1s_not_allowed.py
# User function Template for python3
class Solution:
    def countStrings(self, n):
        # code here

        dp = [[0 for _ in range(n + 1)] for _ in range(2)]
        dp[0][0] = 1
        dp[1][0] = 1
        MOD = int(1e9 + 7)
        for i in range(n):
            dp[0][i + 1] = (dp[0][i] + dp[1][i]) % MOD
            dp[1][i + 1] = dp[0][i]

        return (dp[0][n - 1] + dp[1][n - 1]) % MOD


# {
# Driver Code Starts
# Initial Template for Python 3


# Driver code
if __name__ == "__main__":
    tc = int(input())
    while tc > 0:
        n = int(input())
        ob = Solution()
        ans = ob.countStrings(n)
        print(ans)
        tc = tc - 1
# } Driver Code Ends
